src/dgram.cpp

```diff
@@ -0,0 +1,321 @@
+/*
+    Copyright (c) 2007-2016 Contributors as noted in the AUTHORS file
```

Comment:
Just the current year instead of 2007-2016, since it's a new file. Same in dgram.hpp and tests/test_dgram.cpp. Thanks!